src/classes/GiftBatchStatusGroup.cls

```diff
@@ -56,6 +56,10 @@ public inherited sharing class GiftBatchStatusGroup {
         return status == BDI_DataImport_API.bdiFailed;
     }
 
+    public Boolean isProcessing() {
+        return status == BDI_DataImport_API.bdiProcessing;
+    }
```

Comment:
"isProcessing" seems like a state of the GiftBatch, while this class to me feels like it represents a subset of the Gifts in a batch that all have the same Status.  I'd expect to ask the Batch whether it is processing: "giftBatch.isProcessing()"

There are also now both status and paymentStatus fields on this class, which both seem like properties of the GiftBatch.  

@psadie-sfdo how do you think this class is looking?

Reply:
Yeah, it's clearer to me now that the various "totals" belong to the GiftBatch. I think I'm still ok with GiftBatchStatusGroup having the responsibility of consuming an AggregateResult and determining if the results are in various statuses/payment statuses. GiftBatchTotals then has the responsibility of holding the organized results for a GiftBatch.

_Maybe a tiny detail, but "isProcessing", "isImported", etc in GiftBatchStatusGroup should probably be "hasProcessing" or "hasImported" i.e. "this aggregate result has gifts that are processing" and "this aggregate result has gifts that are imported"._

Ultimately, I think GiftBatchTotals needs to move under and be a member of GiftBatch. Where we're directly referencing GiftBatchTotals in GiftBatchService we should be instantiating a GiftBatch and asking it for its totals.

Reply:
Well said guys! I think we should try to make this change but in a different WI. Adding this to my doc.